src/input/mod.rs

```diff
@@ -253,6 +254,42 @@ impl Input {
         let ok = f(&mut r);
         (ok, r.take_remaining())
     }
+
+    /// Returns `Some(Range)` with the `start` and `end` offsets of `self`
+    /// within the `parent`. `None` is returned if `self` is not within in the
+    /// `parent`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// let parent = dangerous::input(&[1, 2, 3, 4]);
+    /// let sub_range = 1..2;
+    /// let sub = dangerous::input(&parent.as_dangerous()[sub_range.clone()]);
+    ///
+    /// assert_eq!(sub.span_of(parent), Some(sub_range))
+    /// ```
+    pub fn span_of(&self, parent: &Input) -> Option<Range<usize>> {
+        if self.is_within(parent) {
+            let parent_bounds = slice_ptr_range(parent.as_dangerous());
+            let sub_bounds = slice_ptr_range(self.as_dangerous());
+            let start_offset = sub_bounds.start as usize - parent_bounds.start as usize;
+            let end_offset = sub_bounds.end as usize - parent_bounds.start as usize;
+            Some(start_offset..end_offset)
+        } else {
+            None
+        }
+    }
+
+    /// Returns `Some(Range)` with the `start` and `end` offsets of `self`
+    /// within the `parent`. `None` is returned if `self` is not within in the
+    /// `parent` or `self` is empty.
+    pub fn non_empty_span_of(&self, parent: &Input) -> Option<Range<usize>> {
```

Comment:
This one is interesting! Even after an intense 'stare-down' I was unable to see how that is different from `span_of(…)`, which I would expect to also return `None` if the Input is empty and thus it can never contain any other input.
Maybe there is my answer - empty inputs are always contained in an empty parent, unless `non_empty_span_of(…)` is used.

If that's indeed the case, I believe this could be highlighted in the docs.

Reply:
This is something I could highlight better but an empty input slice still can point in or out of the parent. (slice being a fat pointer consisting of a raw ptr and a length). I actually use these semantics with span errors for example [here](https://github.com/avitex/rust-dangerous/blob/master/src/display/writer.rs#L127).

Reply:
I have thought about not having `non_empty_span_of` and just having an empty slice return `None` with `span_of`. The semantics I use and what a consumer use would potentially be different. That being said, storing a `Range<usize>`, is cheaper than a `Option<Range<usize>>` and I expect consumers to use `input.span_of([parent]).unwrap()` which would break empty slices if you care about them. I know I'm being very pedantic here :P

Reply:
@Byron Open to your thoughts on this one :)